### PR TITLE
add starqlte

### PR DESCRIPTION
**Pull request by FakeShell**
### Description

Adds basic support for starqlte including description file and dtb. tested working on starqlte meaning it should also work for starqltechn but i don't have access to that variant so it remains untested

### Checklist

* [x] Have you tested the change you are submitting?
* [x] Is the commits history nice and clean?

### CODE REVIEW

File: Platform/Samsung/sdm845/starqlte.dsc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  SUPPORTED_ARCHITECTURES        = AARCH64
+  BUILD_TARGETS                  = DEBUG|RELEASE
+  SKUID_IDENTIFIER               = DEFAULT
+  FLASH_DEFINITION               = Platform/Qualcomm/sdm845/sdm845.fdf
+  DEVICE_DXE_FV_COMPONENTS       = Platform/Samsung/sdm845/starqlte.fdf.inc
+
+!include Platform/Qualcomm/sdm845/sdm845.dsc
+
+[BuildOptions.common]
+  GCC:*_*_AARCH64_CC_FLAGS = -DENABLE_SIMPLE_INIT -DENABLE_LINUX_SIMPLE_MASS_STORAGE
```
Comment (qaz6750) [inline Platform/Samsung/sdm845/starqlte.dsc:17]: * "It looks like you haven't added a mainline kernel, which means that '- ENABLE_LINUX_SIMPLE_MASS_STORAGE' doesn't need to be added."
* I look forward to seeing its normal operation Xd
* Please also pay attention to Ufs issues

Comment (FakeShell) [reply Platform/Samsung/sdm845/starqlte.dsc:17]: star2qltechn also seems to have that option set
should that be removed as well?

**Comment by FakeShell** [comment]
bump
anyone willing to take a look at this again?

**Comment by BigfootACA** [comment]
> bump anyone willing to take a look at this again?

have you tested linux-simple-mass-storage?

**Comment by FakeShell** [comment]
> > bump anyone willing to take a look at this again?
> 
> have you tested linux-simple-mass-storage?

I have not no.
i took star2qlte/chn as a template for mine and got it booting that way.

**Comment by halal-beef** [comment]
@FakeShell If u have mass storage enabled, make sure u have a mainline dtb

File: Platform/Samsung/sdm845/starqlte.fdf.inc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// per-device BSP DXEs
+
+// ACPI Tables
+
+// Mainline device tree blob
+
```
Comment (BigfootACA) [inline Platform/Samsung/sdm845/starqlte.fdf.inc:6]: enable linux-simple-mass-storage but no mainline device tree blob. you need add mainline device tree blob or disable linux-simple-mass-storage

**Review by BigfootACA — CHANGES_REQUESTED** [review]

File: Platform/Samsung/sdm845/starqlte.fdf.inc (hunk 2 above)
Comment (FakeShell) [reply Platform/Samsung/sdm845/starqlte.fdf.inc:6]: should be good now?

**Review by qaz6750 — APPROVED** [review]
LGTM

**Review by BigfootACA — APPROVED** [review]

**Pull request merged**